src/components/eventsPageComponents/EventList.jsx

```diff
@@ -1,30 +1,77 @@
 import { onSnapshot } from "firebase/firestore";
+import { useRouter } from "next/navigation";
+import { useTranslation } from "next-i18next";
 import React, { useEffect, useState } from "react";
 
+import { useUser } from "@/context/UserContext";
+
+import DateFilter from "./DateFilter";
 import EventCard from "./EventCard";
 import { eventsCollection } from "../../lib/firebase/controller";
 
 function EventList() {
     const [events, setEvents] = useState([]);
-
+    const [filteredEvents, setFilteredEvents] = useState([]);
+    const { user } = useUser();
+    const router = useRouter();
+    // sets the routing according to the user's state
+    const handleUser = () => {
+        user
+            ? router.push("/createEvent")
+            : router.push("/authentication/signUp");
+    };
+    // var to handle translatons
+    const { t } = useTranslation();
+    const onClick = (selectedDate) => {
+        const formattedDate = selectedDate.format("DD MMM YYYY");
+        const filteredEvents = events.filter((event) => {
+            return event.date === formattedDate.toLowerCase();
+        });
+        setFilteredEvents(filteredEvents);
+    };
+    // fetch data from dataBase (fireStore)
```

Comment:
great logic, simple and straight to the point ~this way the filteredeventslist only displays events of that day tho

Reply:
_thanks mate _